DOC: Add syntax highlighting, update pronouns (#19957)
numpy/numpy: doc/source/user/c-info.python-as-glue.rst

advantage of using Fortran to index into multi-dimensional arrays
should be clear from this example.

> .. code-block::

Reviewer: ```suggestion
.. code-block:: fortran
```

Author: There were a few blocks like this in the other docs too which don't seem to take to having syntax highlighting set.

Reviewer: Got it - I was worried that removing the `none` would mean that the code-block would fall back to highlighting the block as if it were Python. I think this *is* the default behavior, though fortunately that doesn't matter for these blocks since everything is all caps :).

Thanks for the follow-up, LGTM then!

Conclusion
----------

> The interface definition file (.pyf) is how you can fine-tune the interface
> between Python and Fortran. There is decent documentation for f2py found on the
> NumPy documentation site at https://numpy.org/devdocs/f2py/index.html. There is
> also more information on using f2py (including how to use it to wrap C codes) at
> https://scipy-cookbook.readthedocs.io under the "Interfacing With Other
> Languages" heading.

Reviewer 2: I'd actually suggest including a label for the f2py main documentation page and referencing it here: 

```suggestion
The interface definition file (.pyf) is how you can fine-tune the interface
between Python and Fortran. There is decent documentation for f2py at
:ref:`f2py`. There is also more information on using f2py (including how to use
it to wrap C codes) at the `"Interfacing With Other Languages" heading of the
SciPy Cookbook.
<https://scipy-cookbook.readthedocs.io/items/idx_interfacing_with_other_languages.html>`_
```

and adding `.. _f2py:` to the top of `doc/source/f2py/index.rst`

These tools have been found useful by others using Python and so are
included here. They are discussed separately because they are
either older ways to do things now handled by f2py, Cython, or ctypes
(SWIG, PyFort) or because of a lack of reasonable documentation (SIP, Boost).
Links to these methods are not included since the most relevant
can be found using Google or some other search engine, and any links provided
here would be quickly dated. Do not assume that inclusion in this list means
that the package deserves attention. Information about these packages are

> collected here because many people have found them useful and I'd like to give

Reviewer 2: ```suggestion
collected here because many people have found them useful and we'd like to give
```

classes and functions to Python. The amazing part of the Boost.Python
approach is that it works entirely in pure C++ without introducing a
new syntax. Many users of C++ report that Boost.Python makes it
possible to combine the best of both worlds in a seamless fashion. Using Boost

> to wrap simple C-subroutines is usually over-kill. It's primary purpose is to

Reviewer 2: ```suggestion
to wrap simple C-subroutines is usually over-kill. Its primary purpose is to
```
